pkg/dashboard/dashboard.go

```diff
@@ -150,11 +179,71 @@ func (ws *Server) ListenAndServe(ctx context.Context) error {
 		).Error("failed to parse management API URL")
 		panic("failed to parse management API URL")
 	}
-	router.Any("/opni-api/*any", gin.WrapH(http.StripPrefix("/opni-api", httputil.NewSingleHostReverseProxy(mgmtUrl))))
+	apiGroup := router.Group("/opni-api")
+	apiGroup.Use(middleware.Handler())
+	apiGroup.Any("/*any", gin.WrapH(http.StripPrefix("/opni-api", httputil.NewSingleHostReverseProxy(mgmtUrl))))
+
+	proxy := router.Group("/proxy")
+	proxy.Use(middleware.Handler())
+	ws.pl.Hook(hooks.OnLoad(func(p types.ProxyPlugin) {
+		log := lg.WithGroup("proxy")
+		pluginRouter, err := proxyrouter.NewRouter(proxyrouter.RouterConfig{
+			Store:  ws.ds.StorageBackend(),
+			Logger: log,
+			Client: p,
+		})
+		if err != nil {
+			log.With(
+				logger.Err(err),
+			).Error("failed to create plugin router")
+			return
+		}
```

Comment:
should this be a panic?

Reply:
I don't think so.  This follows the standard pattern where if we're unable to load a plugin we make the best effort to continue without taking down the whole system.